org.envirocar.map/src/main/java/org/envirocar/map/provider/mapbox/MapboxMapController.kt

```diff
@@ -0,0 +1,266 @@
+        )
+    }
+
+    override fun notifyCameraUpdate(cameraUpdate: CameraUpdate, animation: Animation?) = runWhenReady {
+        super.notifyCameraUpdate(cameraUpdate, animation)
+
+        fun setOrEaseCamera(cameraOptions: CameraOptions) {
+            if (animation == null) {
+                viewInstance.mapboxMap.setCamera(cameraOptions)
+            } else {
+                viewInstance.mapboxMap.easeTo(
+                    cameraOptions,
+                    MapAnimationOptions.Builder()
+                        .duration(animation.duration)
+                        .build()
+                )
+            }
+        }
+
+        with(cameraUpdate) {
+            when (this) {
+                is CameraUpdate.Companion.CameraUpdateBasedOnBounds -> {
+                    viewInstance.mapboxMap.cameraForCoordinates(
+                        points.map { it.toMapboxPoint() },
+                        CameraOptions.Builder()
+                            .padding(padding.toDouble().let { EdgeInsets(it, it, it, it) })
+                            .build(),
+                        null,
+                        null,
+                        null
+                    ) {
+                        setOrEaseCamera(it)
+                    }
+                }
+
+                is CameraUpdate.Companion.CameraUpdateBasedOnPoint -> {
+                    setOrEaseCamera(
+                        CameraOptions.Builder()
+                            .center(point.toMapboxPoint())
+                            .build()
+                    )
+                }
+
+                is CameraUpdate.Companion.CameraUpdateBearing -> {
+                    setOrEaseCamera(
+                        CameraOptions.Builder()
+                            .bearing(bearing.toMapboxBearing())
+                            .build()
+                    )
+                }
+
+                is CameraUpdate.Companion.CameraUpdateTilt -> {
+                    // Mapbox SDK refers to tilt as pitch.
+                    setOrEaseCamera(
+                        CameraOptions.Builder()
+                            .pitch(tilt.toMapboxTilt())
+                            .build()
+                    )
+                }
+
+                is CameraUpdate.Companion.CameraUpdateZoom -> {
+                    setOrEaseCamera(
+                        CameraOptions.Builder()
+                            .zoom(zoom.toMapboxZoom())
+                            .build()
+                    )
+                }
+            }
+        }
+    }
+
+    override fun addMarker(marker: Marker) = runWhenReady {
+        if (markers.contains(marker.id)) {
+            error("Marker with ID ${marker.id} already exists.")
+        }
+        var options = PointAnnotationOptions()
+        marker.point.also {
+            options = options.withPoint(it.toMapboxPoint())
+        }
+        marker.title?.also {
+            options = options.withTextField(it)
+        }
```

Comment:
Wouldn't `let` instead of `also` be sufficient, since you don't reuse the object to apply another funtion on it?